cygnss_wetlands/animation/animate.py

```diff
@@ -0,0 +1,346 @@
+
+    return figure_month_path
+
+
+def createMonthlyIntervals(year: int, month: int, numIntervals: int):
+    """
+    Outputs a list of intervals for the given month
+
+    Args:
+        year (int): The year you wish to create monthly intervals for
+        month (int): The month you wish to create monthly intervals for
+        numIntervals (int): The number of intervals you want in the month
+
+    Returns:
+        list of tuples: A list of tuples of length numIntervals containing non-overlapping dates for the given month
+
+    Example:
+        >>> intervals = createMonthlyIntervals(year=2023, month=1, numIntervals=2)
+        >>> print(intervals)
+        [(1, 15), (16, 31)]
+    """
+    monthIntervals = []
+    numDays = calendar.monthrange(year, month)[1]
+    daysPerInterval = numDays // numIntervals
+    for i in range(numIntervals):
+        intervalStart = 1 + (i * daysPerInterval)
+        intervalEnd = (i + 1) * daysPerInterval
+        if i == numIntervals - 1:
+            intervalEnd += numDays % numIntervals
+        monthIntervals.append((intervalStart, intervalEnd))
+    return monthIntervals
+
+
+def generateFigure(
+    figureName: str,
+    variableName: str,
+    year: int,
+    month: int,
+    startDate: int,
+    endDate: int,
+    grid: GridType,
+    max: float,
+    min: float,
+):
+    """
+    Creates and saves a figure to figureName for the given year, month, and date range on the prescribed grid.
+
+    Args:
+        figureName (str): The filename to save the figure to
+        variableName (str): Variable to plot
+        year (int): The year in numerical format the figure is representing
+        month (int): The month in numerical format the figure is representing
+        startDate (int): The start date of the month in numerical format the figure is representing
+        endDate (int): The end date of the month in numerical format the figure is representing
+        grid (GridType): The EASE2 GridType to bin data in for the figure
+        max (float): The maximum value of the colorbar
+        min (float): The minimum value of the colorbar
+
+    Returns:
+        None: Saves figure to figureName
+    """
+    d1 = datetime.datetime(year, month, startDate)
+    d2 = datetime.datetime(year, month, endDate)
+
+    snr = reader.aggregate(variable_name=variableName, grid=grid, start_date=d1, end_date=d2)
+
+    # Plot
+    bbox_grid_xmin, bbox_grid_ymin = grid.lonlat2cr(reader.xmin, reader.ymin)
+    bbox_grid_xmax, bbox_grid_ymax = grid.lonlat2cr(reader.xmax, reader.ymax)
+
+    fig, ax = plt.subplots(figsize=(6, 6))
+
+    cmap = plt.get_cmap("viridis")
+    no_data_color = "gray"
+    cmap.set_bad(color=no_data_color)
+
+    pos = ax.imshow(snr, cmap=cmap, vmin=min, vmax=max)
+    plt.colorbar(pos, ax=ax, label=variableName.upper())
+
+    # Create a legend for missing data
+    handles = [plt.Rectangle((0, 0), 1, 1, color="gray")]
+    labels = ["No Data"]
+    plt.legend(handles, labels)
+
```

Comment:
For plotting raster maps in matplotlib -- the `cartopy` package makes this a lot easier -- something to consider in the future (we can add as a TODO) ? (e.g. https://scitools.org.uk/cartopy/docs/latest/gallery/gridlines_and_labels/gridliner.html#sphx-glr-gallery-gridlines-and-labels-gridliner-py)